Added python3-build

m-elwin opened the pull request:
Please add the following dependency to the rosdep database.

## Package name:

python3-build

## Package Upstream Source:

https://github.com/pypa/build

## Purpose of using this:

This dependency is being used for a package that makes it easier to add python code to an  ament_cmake project.
It may be use to others who want to work on various python-related build processes with ROS.

Distro packaging links:

## Links to Distribution Packages

<!-- Replace the REQUIRED areas with the URL to the package.  For IF AVAILABLE areas, either put in the URL to the package or state 'not available'.
More info at https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md#guidelines-for-rosdep-rules -->

- Debian: https://packages.debian.org/
  - https://packages.debian.org/bullseye/main/python3-build
- Ubuntu: https://packages.ubuntu.com/
  - https://packages.ubuntu.com/jammy/python3-build
  - Not available in focal or earlier, added pip dependency
- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/python-build/python3-build/
- Arch: https://www.archlinux.org/packages/
  - https://archlinux.org/packages/extra/any/python-build/
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/dev-python/build
- macOS: https://formulae.brew.sh/
  - https://formulae.brew.sh/formula/python-build
- Alpine: https://pkgs.alpinelinux.org/packages
  - https://pkgs.alpinelinux.org/package/edge/community/x86_64/py3-build
- NixOS/nixpkgs: https://search.nixos.org/packages
  - OPTIONAL
- openSUSE: https://software.opensuse.org/package/
  - IF AVAILABLE
- rhel: https://rhel.pkgs.org/
  - IF AVAILABLE

m-elwin committed:
Added python3-build

audrow commented:
@m-elwin, it looks like the package is available for RHEL, too. Would you mind adding it?

m-elwin commented:
@audrow added for rhel.

nuclearsandwich commented on rosdep/python.yaml:
```diff
@@ -5104,6 +5104,23 @@ python3-bson:
+    rhel9: [python3-build]
+  ubuntu:
+    focal:
+      pip:
+        packages: [build]
```
Unless you specifically need the pip packages on Focal it's better not to add them.

```suggestion
```

nuclearsandwich commented on rosdep/python.yaml:
```diff
@@ -5104,6 +5104,23 @@ python3-bson:
+  gentoo: [dev-python/build]
+  osx: [python-build]
+  rhel:
+    rhel8:
+      pip:
+        packages: [build]
```
Unless you specifically need the pip packages for RHEL 8 (Thanks for adding RHEL 9 in response to earlier feedback!) it's better not to add them.

```suggestion
```

nuclearsandwich reviewed (commented):
I'd like to drop the pip packages unless they're explicitly needed by a contributor in order to reduce the use of mixed pip/platform keys overall. If these pip keys are in use by the contributor in transition, then they don't need to be removed. But I'm against adding them without a concrete use case.

m-elwin commented:
Switched to '9' for rhel and removed pip dependencies in focal/rhel8. This package is not available via apt/yum for Focal/rhel8 as far as I can tell, but I also don't know of a need for this dependency on those platforms.

clalancette commented on rosdep/python.yaml:
```diff
@@ -5104,6 +5104,17 @@ python3-bson:
+  osx: [python-build]
+  rhel:
+    '9': [python3-build]
+  ubuntu:
+    jammy: [python3-build]
```
Generally it is better to mark things as '*' so that they are automatically used for newer versions, and then to "null" out the older ones that aren't supported.  So in this case this would be better:

```suggestion
  ubuntu:
    '*': [python3-build]
    focal: null
```

m-elwin replied:
Fixed

clalancette commented on rosdep/python.yaml:
```diff
@@ -5104,6 +5104,17 @@ python3-bson:
+  fedora: [python3-build]
+  gentoo: [dev-python/build]
+  osx: [python-build]
+  rhel:
+    '9': [python3-build]
```
Same here:

```suggestion
  rhel:
    '*': [python3-build]
    '8': null
```

m-elwin replied:
Fixed
